platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/MigrationReportImpl.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    /**
+     * Creates a new migration report by transferring the errors and warnings from the provided report
+     * as warnings for this report. Transfer also the start time.
+     *
+     * @param operation the type of migration operation for this report
+     * @param report    the report to be transferred over
+     * @param consumer  an optional consumer to call whenever a new migration message is recorded
+     *                  during the operation
+     * @throws IllegalArgumentException if <code>operation</code> or <code>report</code> is <code>null</code>
+     */
+    public MigrationReportImpl(MigrationOperation operation, MigrationReportImpl report,
+            Optional<Consumer<MigrationMessage>> consumer) {
+        Validate.notNull(operation, "invalid null operation");
+        Validate.notNull(report, "invalid null report");
+        report.runCodes(); // to get all errors and warnings recorded
+        this.messages = report.messages.stream()
+                .map(Messages::downgradeToWarning)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toCollection(LinkedHashSet::new)); // LinkedHashSet to prevent duplicate and maintain order
+        this.numWarnings = messages.size();
+        this.operation = operation;
+        this.consumer = consumer;
+        this.start = report.getStartTime();
+    }
+
+    @Override
+    public MigrationOperation getOperation() {
+        return operation;
+    }
+
+    @Override
+    public Instant getStartTime() {
+        return start;
+    }
+
+    @Override
+    public Optional<Instant> getEndTime() {
+        return end;
+    }
+
+    @Override
+    public MigrationReportImpl record(MigrationMessage msg) {
+        Validate.notNull(msg, "invalid null message");
+        final String level;
+
+        if (msg instanceof MigrationException) {
+            this.numErrors++;
+            level = "error";
+        } else if (msg instanceof MigrationWarning) {
+            this.numWarnings++;
+            level = "warning";
+        } else if (msg instanceof MigrationInformation) {
+            this.numInfos++;
+            level = "info";
+        } else {
+            level = "message";
+        }
+        LOGGER.debug("migration {}: {}", level, msg);
+        messages.add(msg);
+        consumer.ifPresent(c -> c.accept(msg));
+        return this;
+    }
+
+    @Override
+    public MigrationReport doAfterCompletion(Consumer<MigrationReport> code) {
+        Validate.notNull(code, "invalid null code");
+        codes.add(code);
+        return this;
+    }
+
+    @Override
+    public Stream<MigrationMessage> messages() {
+        return messages.stream();
+    }
+
+    @Override
+    public boolean wasSuccessful() {
+        runCodes();
+        return (numErrors == 0);
+    }
+
+    @Override
+    public boolean wasSuccessful(Runnable code) {
```

Comment:
The documentation no longer documents which method calls the code. These methods might not be named properly since they might cause confusion with the wasSuccessful(). They do not do the same thing. wasSuccessful() check if the whole report was successful whereas these only returns true or false based on wether or not the code executed is successful.